connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceConnectorConfig.java

```diff
@@ -17,10 +17,16 @@
 
 package org.apache.kafka.connect.runtime;
 
+import org.apache.kafka.common.config.ConfigDef;
+
 import java.util.Map;
 
 public class SourceConnectorConfig extends ConnectorConfig {
 
+    public static ConfigDef configDef() {
+        return config;
```

Comment:
This doesn't seem right. If we return `config` here, it will pull it from `ConnectorConfig`. But the same is true of both `ConnectorConfig.configDef()` and `SinkConnectorConfig.configDef()`. Since its all just static data, we'll only have one instance, and when `SinkConnectorConfig` is initialized it'll define the `topics` configuration and that will cause it to be added to *every* `XConnectorConfig` class's configuration.

I think we need to make sure we have a way to isolate the different instances from each other. That is why Confluent's rest-utils uses a `baseConfig()` method that recreates the entire config from scratch: https://github.com/confluentinc/rest-utils/blob/master/core/src/main/java/io/confluent/rest/RestConfig.java#L97